service/controller/v13/resource/instance/create_worker_instances_upgrading.go

```diff
@@ -0,0 +1,84 @@
+package instance
+
+import (
+	"context"
+	"fmt"
+
+	corev1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/core/v1alpha1"
+	"github.com/giantswarm/azure-operator/service/controller/v13/key"
+	"github.com/giantswarm/azure-operator/service/controller/v13/resource/instance/internal/state"
+	"github.com/giantswarm/microerror"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func (r *Resource) workerInstancesUpgradingTransition(ctx context.Context, obj interface{}, currentState state.State) (state.State, error) {
+	customObject, err := key.ToCustomObject(obj)
+	if err != nil {
+		return "", microerror.Mask(err)
+	}
+
+	versionValue := map[string]string{}
```

Comment:
is this the kind of information that we could put in the context?

Reply:
Theoretically, but when it's already available in `Status` of reconciled CR I feel that would be sort of duplicate information.

What kind of benefit do you see in that approach?

Reply:
I believe I miss clicked. I was talking about the drainerConfigs. Instead of repeating the logic to fetch that information, maybe we could put it in the context, and just read it from there in here. That would avoid duplication, but I don't see it as a big problem as it currently is.

Reply:
In `instance` resource's case I would prefer to not do that because those `DrainerConfigs` are only needed in two phases (`MasterInstancesUpgrading` and `WorkerInstancesUpgrading`) during upgrade and otherwise they aren't needed.

If they were needed during every reconciliation round, then that would be a good idea.